pkg/resource/stack/deployment.go

```diff
@@ -120,10 +125,24 @@ func SerializeDeployment(snap *deploy.Snapshot, sm secrets.Manager, showSecrets
 		enc = config.NewPanicCrypter()
 	}
 
+	aliases := make(aliases)
+	for _, state := range snap.Resources {
+		// Add to aliased maps
+		for _, alias := range state.Aliases {
+			// For ease of implementation, some SDKs may end up creating the same alias to the
+			// same resource multiple times.  That's fine, only error if we see the same alias,
+			// but it maps to *different* resources.
+			if otherUrn, has := aliases[alias]; has && otherUrn != state.URN {
+				return nil, fmt.Errorf("Two resources ('%s' and '%s') aliased to the same: '%s'", otherUrn, state.URN, alias)
+			}
+			aliases[alias] = state.URN
+		}
+	}
+
```

Comment:
Is it possible to pull this into a method? Perhaps something like `snap.GenerateAliasesMap()`. Separating the logic might make it easier to unit test, and it appears this block is total.